src/main/java/io/vertx/core/json/jackson/HybridJacksonPool.java

```diff
@@ -0,0 +1,208 @@
+
+    private static final int CACHE_LINE_SHIFT = 4;
+
+    private static final int CACHE_LINE_PADDING = 1 << CACHE_LINE_SHIFT;
+
+    private final XorShiftThreadProbe threadProbe;
+
+    private final AtomicReferenceArray<Node> heads;
+
+    public StripedLockFreePool(int stripesCount) {
+      if (stripesCount <= 0) {
+        throw new IllegalArgumentException("Expecting a stripesCount that is larger than 0");
+      }
+
+      int size = roundToPowerOfTwo(stripesCount);
+      this.heads = new AtomicReferenceArray<>(size * CACHE_LINE_PADDING);
+
+      int mask = (size - 1) << CACHE_LINE_SHIFT;
+      this.threadProbe = new XorShiftThreadProbe(mask);
+    }
+
+    @Override
+    public BufferRecycler acquirePooled() {
+      int index = threadProbe.index();
+
+      Node currentHead = heads.get(index);
+      while (true) {
+        if (currentHead == null) {
+          return new VThreadBufferRecycler(index);
+        }
+
+        if (heads.compareAndSet(index, currentHead, currentHead.next)) {
+          currentHead.next = null;
+          return currentHead.value;
+        } else {
+          currentHead = heads.get(index);
+        }
+      }
+    }
+
+    @Override
+    public void releasePooled(BufferRecycler recycler) {
+      VThreadBufferRecycler vThreadBufferRecycler = (VThreadBufferRecycler) recycler;
+      Node newHead = new Node(vThreadBufferRecycler);
+
+      Node next = heads.get(vThreadBufferRecycler.slot);
+      while (true) {
+        if (heads.compareAndSet(vThreadBufferRecycler.slot, next, newHead)) {
+          newHead.next = next;
+          return;
+        } else {
+          next = heads.get(vThreadBufferRecycler.slot);
+        }
+      }
+    }
+
+    private static class Node {
+      final VThreadBufferRecycler value;
+      Node next;
+
+      Node(VThreadBufferRecycler value) {
+        this.value = value;
+      }
+    }
+  }
+
+  private static class VThreadBufferRecycler extends BufferRecycler {
+    private final int slot;
+
+    VThreadBufferRecycler(int slot) {
+      this.slot = slot;
+    }
+  }
+
+  private static class VirtualPredicate {
+    private static final MethodHandle virtualMh = findVirtualMH();
+
+    private static MethodHandle findVirtualMH() {
+      try {
+        return MethodHandles.publicLookup().findVirtual(Thread.class, "isVirtual",
+          MethodType.methodType(boolean.class));
+      } catch (Exception e) {
+        return null;
+      }
+    }
+
+    private static Predicate<Thread> findIsVirtualPredicate() {
+      if (virtualMh != null) {
+        return new Predicate<Thread>() {
+          @Override
+          public boolean test(Thread thread) {
+            try {
+              return (boolean) virtualMh.invokeExact(thread);
+            } catch (Throwable e) {
+              throw new RuntimeException(e);
+            }
+          }
+        };
+      }
+
+      return new Predicate<Thread>() {
```

Comment:
why not simply: `return thread->false;`

Reply:
We decided to avoid using lambdas here for performance reasons.